### PR TITLE
Connection builder simplification

### PR DESCRIPTION
**Pull request by TwistedTwigleg**
*Description of changes:*

Make a few small changes to reduce the boiler plate needed for creating an MQTT connection using the MQTT builder without breaking backwards compatibility. Now it is possible to make a connection without needing to manually create a HostResolver, EventLoopGroup, or ClientBootstrap, this can all be done automatically by the builder (and is automatically freed when done).

For RawPubSub and Greengrass samples, both are simplified but since they do not use the builder, they are not quite as simplified as other samples. However, now they do not manually create the EventLoopGroup, HostResolver, or ClientBootstrap (instead using a default static one), but because the builder wasn't used, an additional line of code is needed to free the static ClientBootstrap prior to calling `CrtResource.releaseStaticDefault` to free resources.

Note: Requires the changes in [This PR on the CRT repository](https://github.com/awslabs/aws-crt-java/pull/443) in order to function and build correctly.

___________

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.*

### CODE REVIEW

**Comment by chong-zi** [comment]
您好，您发给我的邮件我已收到。                                黄继龙

File: samples/Pkcs11PubSub/src/main/java/pkcs11pubsub/Pkcs11PubSub.java [hunk 1]
```diff
@@ -203,17 +203,15 @@ public void onConnectionResumed(boolean sessionPresent) {
...
+            try (
                     AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder
                             .newMtlsPkcs11Builder(pkcs11Options)) {
 
                 if (rootCaPath != null) {
                     builder.withCertificateAuthorityFromPath(null, rootCaPath);
                 }
 
-                builder.withBootstrap(clientBootstrap).withConnectionEventCallbacks(callbacks).withClientId(clientId)
+                builder.withBootstrap().withConnectionEventCallbacks(callbacks).withClientId(clientId)
```
Comment (bretambrose) [inline samples/Pkcs11PubSub/src/main/java/pkcs11pubsub/Pkcs11PubSub.java:214]: remove withBootstrap

File: samples/PubSubStress/src/main/java/pubsubstress/PubSubStress.java [hunk 2]
```diff
@@ -329,15 +326,13 @@ public static void main(String[] args) {
...
-                HostResolver resolver = new HostResolver(eventLoopGroup);
-                ClientBootstrap clientBootstrap = new ClientBootstrap(eventLoopGroup, resolver);
+            try (
                 AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder.newMtlsBuilderFromPath(certPath, keyPath)) {
 
                 builder.withCertificateAuthorityFromPath(null, rootCaPath)
                     .withEndpoint(endpoint)
                     .withCleanSession(true)
-                    .withBootstrap(clientBootstrap)
+                    .withBootstrap()
```
Comment (bretambrose) [inline samples/PubSubStress/src/main/java/pubsubstress/PubSubStress.java:335]: remove

File: sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java [hunk 3]
```diff
@@ -390,6 +390,15 @@ public AwsIotMqttConnectionBuilder withBootstrap(ClientBootstrap bootstrap) {
         return this;
     }
 
+    /**
+     * Configures and uses the static default bootstrap, event loop, and resolver for this builder
+     * @return
+    */
+    public AwsIotMqttConnectionBuilder withBootstrap() {
```
Comment (bretambrose) [inline sdk/src/main/java/software/amazon/awssdk/iot/AwsIotMqttConnectionBuilder.java:397]: Remove and update build() function to use default bootstrap if necessary.

File: samples/Shadow/src/main/java/shadow/ShadowSample.java [hunk 4]
```diff
@@ -237,7 +232,7 @@ public void onConnectionResumed(boolean sessionPresent) {
                     .withEndpoint(endpoint)
                     .withCleanSession(true)
                     .withConnectionEventCallbacks(callbacks)
-                    .withBootstrap(clientBootstrap);
+                    .withBootstrap();
```
Comment (bretambrose) [inline samples/Shadow/src/main/java/shadow/ShadowSample.java:235]: remove

**Review by bretambrose — CHANGES_REQUESTED** [review]

File: samples/BasicPubSub/src/main/java/pubsub/PubSub.java [hunk 5]
```diff
@@ -257,16 +255,14 @@ public void onConnectionResumed(boolean sessionPresent) {
...
-            ClientBootstrap clientBootstrap = new ClientBootstrap(eventLoopGroup, resolver);
+        try (
             AwsIotMqttConnectionBuilder builder = AwsIotMqttConnectionBuilder.newMtlsBuilderFromPath(certPath, keyPath)) {
 
             if (rootCaPath != null) {
                 builder.withCertificateAuthorityFromPath(null, rootCaPath);
             }
 
-            builder.withBootstrap(clientBootstrap)
+            builder.withBootstrap()
```
[the remark this answers is not in the document]
Comment (TwistedTwigleg) [reply samples/BasicPubSub/src/main/java/pubsub/PubSub.java:265]: I just tested and it seems like it cannot.
I probably missing something in the CRT PR. I will take a look and see if I can figure it out.

[the remark this answers is not in the document]
Comment (TwistedTwigleg) [reply samples/BasicPubSub/src/main/java/pubsub/PubSub.java:265]: Found it - there's a validation check in the V2 SDK I missed that causes it not to work. Modifying it fixes the issue and allows removing `withBootstrap`. 👍

**Comment by TwistedTwigleg** [comment]
Thanks for the review! I modified the code accordingly

**Review by sbSteveK — APPROVED** [review]
Merge with latest changes to main, pass tests, and you're good.

**Pull request merged**